docs/specs/sequences/endpoint_reconfiguration.md

```diff
@@ -0,0 +1,112 @@
+[**UDMI**](../../../) / [**Docs**](../../) / [**Specs**](../) / [**Sequences**](./) / [Endpoint Reconfiguration](#)
+
+# Endpoint Reconfiguration
+
+Endpoint reconfiguration is the reconfiguration of the UDMI endpoint through the
+UDMI blob delivery mechanisms via UDMI config messages.
+
+The [endpoint configuration blob](https://github.com/faucetsdn/udmi/blob/master/tests/configuration_endpoint.tests/simple.json) is a JSON object defined by
+[configuration_endpoint.json](https://faucetsdn.github.io/udmi/gencode/docs/configuration_endpoint.html), which is base64 encoded in the config message.
+
+
+## Tests
+
+### Valid Endpoint (Successful) Reconfiguration
+
+```mermaid
+%%{wrap}%%
+sequenceDiagram
+    autonumber
+    participant D as Device
+    participant E as Original Endpoint
+    participant E' as New Endpoint
+    E->>D:CONFIG MESSAGE<br>blobset.blobs._iot_endpoint_config.blob = <ENDPOINT><br>blobset.blobs._iot_endpoint.blob.phase = "final"
+    D->>E:STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.blob.phase = "apply"
+    D-->>E':CONNECTION ATTEMPT
+    E'-->>D:SUCCESS
+    E'->>D:CONFIG
+    D->>E':STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.blob.phase = "final"
+    note over D: Reboot device
+    D-->>E':CONNECTION ATTEMPT
+    E'-->>D:SUCCESS
+    E'->>D:CONFIG
+```
+
+### Invalid Endpoint (Unsuccessful Reconfiguration)
+
+```mermaid
+%%{wrap}%%
+sequenceDiagram
+    autonumber
+    participant D as Device
+    participant E as Original Endpoint
+    participant E' as New Endpoint
+    E->>D:CONFIG MESSAGE<br>blobset.blobs._iot_endpoint_config.blob = <ENDPOINT><br>blobset.blobs._iot_endpoint.blob.phase = "final"
+    D->>E:STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.blob.phase = "preparing"
```

Comment:
I don't understate this state. Would we not also see this in the success state?

Reply:
Trevor noted it should be apply rather than preparing, but I didn't realise I only changed one of the diagrams. So yes, both would enter the `apply` phase after receiving an endpoint configuration - if that's answered your question